pkgs/samples/lib/google_ai.dart

```diff
@@ -157,19 +159,40 @@ class _ChatWidgetState extends State<ChatWidget> {
 
   @override
   Widget build(BuildContext context) {
-    final history = _chat.history.toList();
+    var textFieldDecoration = InputDecoration(
+      contentPadding: const EdgeInsets.all(15),
+      hintText: 'Enter a prompt...',
+      border: OutlineInputBorder(
+        borderRadius: const BorderRadius.all(
+          Radius.circular(14),
+        ),
+        borderSide: BorderSide(
+          color: Theme.of(context).colorScheme.secondary,
+        ),
+      ),
+      focusedBorder: OutlineInputBorder(
+        borderRadius: const BorderRadius.all(
+          Radius.circular(14),
+        ),
+        borderSide: BorderSide(
+          color: Theme.of(context).colorScheme.secondary,
+        ),
+      ),
+    );
+
     return Padding(
       padding: const EdgeInsets.all(8.0),
       child: Column(
         mainAxisAlignment: MainAxisAlignment.center,
         crossAxisAlignment: CrossAxisAlignment.start,
         children: [
           Expanded(
-            child: ListView.builder(
+            child: widget.apiKey.isNotEmpty
+                ? ListView.builder(
               controller: _scrollController,
               itemBuilder: (context, idx) {
-                final content = history[idx];
-                final text = content.parts
+                var content = _chat.history.toList()[idx];
```

Comment:
As a note, I changed this piece intentionally to avoid recreating the same list, and potentially reevaluating the `history` iterable for each item, but I'll open an upstream fix so it's not lost.